### PR TITLE
4989 - search x header alignment fixes

### LINKED ISSUE #4989: Header: larger search x icon alignment issue
**Describe the bug**
The x icon of Searchfield was misaligned 

**To Reproduce**

1. Go to https://master-enterprise.demo.design.infor.com/components/header/example-buttons-larger-search.html
2. Enter text in searchfield
3. See error
4. X icon is not properly aligned

**Expected behavior**
should be properly aligned

**Version**

- ids-enterprise: v4.39.0-dev

**Screenshots**
image

**Platform**
All

**Additional context**

### PR DESCRIPTION
**Pull request by rob2d**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Should solve most (if not all) of the styling issues with the search fields and toolbar search fields that contain clear buttons.

**Related github/jira issue (required)**:
Closes #4989 

**Steps necessary to review your pull request (required)**:
- Build the demo application and look over components in the places listed below. 
- Type in some content and see an "X" to clear content. 

Previously the "X" would be down into the search field on certain scenarios; this should no longer happen. 

As mentioned in the issue, the components in these sections should be checked locally.
[http://localhost:4000/components/searchfield](http://localhost:4000/components/searchfield)
[http://localhost:4000/components/toolbarsearchfield](http://localhost:4000/components/toolbarsearchfield)
[http://localhost:4000/components/header](http://localhost:4000/components/header)
[http://localhost:4000/components/tabs-module](http://localhost:4000/components/tabs-module)

For reference/sanity check if the PR is merged:
[https://master-enterprise.demo.design.infor.com/components/searchfield](https://master-enterprise.demo.design.infor.com/components/searchfield)
[https://master-enterprise.demo.design.infor.com/components/toolbarsearchfield](https://master-enterprise.demo.design.infor.com/toolbarsearchfield)
[https://master-enterprise.demo.design.infor.com/components/header](https://master-enterprise.demo.design.infor.com/components/header)
[https://master-enterprise.demo.design.infor.com/components/tabs-module](https://master-enterprise.demo.design.infor.com/components/tabs-module)

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [X] A note to the change log.

### CODE REVIEW

**Comment by rob2d** [comment]
Just a note that also in this branch/PR:

- TESTING.md changes: minor clarifications/updates, added a nav/TOC since there's non-sequential topics
- refactored/iterated on specific data grid e2e tests since those were flaking out here and there w some new conventions where we can avoid sleep in this file going forward.

File: docs/TESTING.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
 
 The IDS components are backed by both functional and end-to-end (e2e) test suites.  When contributing to the IDS enterprise project, before we can accept pull requests we expect that new tests will be provided to prove that new functionality works, and that all existing tests pass.
 
+- [Test Stack](#test-stack)
```
Comment (tmcconechy) [inline docs/TESTING.md:5]: I think we did pretty much the same thing here would you try and put these on https://github.com/infor-design/enterprise/pull/5023 or we might have a merge conflict? thanks

Comment (rob2d) [reply docs/TESTING.md:5]: and yep we did have some conflicts but sorted it out and tried to preserve your changes on it.

**Review by tmcconechy — APPROVED** [review]

File: src/components/searchfield/_searchfield.scss [hunk 2]
```diff
@@ -320,8 +324,12 @@
     }
 
     button.close {
-      right: 35px;
-      top: 4px;
+      align-items: center;
+      display: flex;
```
Comment (EdwardCoyle) [inline src/components/searchfield/_searchfield.scss:328]: I found one Searchfield example where the close button is still displayed when the input has no contents:  http://localhost:4000/components/tabs-module/example-category-searchfield-go-button.html -- seems to be tied to this rule?  We may have to qualify it somehow with `:empty` or with `.is-empty` coming from the wrapper?

**Review by EdwardCoyle — CHANGES_REQUESTED** [review]
Great PR with all the docs improvements, thanks for this!  Just found one thing below

File: src/components/searchfield/_searchfield.scss (hunk 2 above)
Comment (rob2d) [reply src/components/searchfield/_searchfield.scss:328]: Good catch Ed, thanks. Checking into it.

Comment (rob2d) [reply src/components/searchfield/_searchfield.scss:328]: forgot to push there -- just running through CI now

**Review by tmcconechy — APPROVED** [review]

**Review by EdwardCoyle — DISMISSED** [review]

**Pull request merged**